go/sqltypes/value.go

```diff
@@ -235,6 +237,13 @@ func (v Value) ToBytes() []byte {
 	if v.typ == Expression {
 		return nil
 	}
+	if v.typ == HexVal {
+		dv, err := v.decodeHexVal()
+		if err != nil {
+			log.Errorf("Unexpected error seen when returning MySQL representation of SQL Hex value: %v", err)
+		}
+		return dv
```

Comment:
if not able to decode hex val, then should it not return error then returning dv (which will contain nil value).

Reply:
Yeah, I feel like we should... but that method never returned an error so for now I just logged it. If you think it's OK to modify all the `Value.ToBytes()` call sites to handle errors then I'll do that.

Reply:
I think/hope this is what you mean/wanted: https://github.com/vitessio/vitess/pull/9277/commits/99d065a3f6e2c8d46f6672c37f604d6d4686d079

If not, at least we can easily revert the single commit.